Neo4jClient/Cypher/CypherFluentQuery.cs

```diff
@@ -148,17 +148,23 @@ protected ICypherFluentQuery<TResult> Mutate<TResult>(Action<QueryWriter> callba
 
         public ICypherFluentQuery WithParam(string key, object value)
         {
+            if (string.IsNullOrWhiteSpace(key) || char.IsDigit(key[0]) || char.IsSymbol(key[0]) || key[0] == '{')
+                throw new ArgumentException("Parameters may consist of letters and numbers, and any combination of these, but cannot start with a number or a currency symbol.", key);
+
             if (QueryWriter.ContainsParameterWithKey(key))
-                throw new ArgumentException("A parameter with the given key is already defined in the query.", nameof(key));

```

Comment:
Hi, I think you are wrong. The parameter name of the WithParam method is "key" and that is the reason this exception always told the error would be in "Parameter name: key". With my changes, the parameter name used in the query is put there as it is the value of the method parameter key.

In the version of my PR correct output is given (instead of parameter "key" as in current version):
![image](https://user-images.githubusercontent.com/14079908/214034713-5efa36ee-3fd1-4d7f-95e4-4c78a866ec90.png)

Reply:
Just change the parameter name in lines 62 and 63 CypherFluentQueryWithParamTests.cs from "key" to "foo" and you'll see that your test will break as this only works with the word "key" because the method parameter also is named key.